packages/store/addon/-private/ts-interfaces/minimum-serializer-interface.ts

```diff
@@ -0,0 +1,313 @@
+  by `EmberData`), while data sent to an API is `"serialized"`
+  into the format the API expects.
+
+  ### Implementing a Serializer
+
+  There are only two required serializer methods, one for
+  normalizing data from the server API format into `JSON:API`, and
+  another for serializing records via `Snapshot`s into the expected
+  server API format.
+
+  To implement a serializer, export a class implementing the
+  [MinimumSerializerInterface](MinimumSerializerInterface) from
+  the `app/serializers/` directory. An example is below.
+
+  ```ts
+  import EmberObject from '@ember/object';
+
+  export default class ApplicationSerializer extends EmberObject {
+    normalizeResponse(store, schema, rawPayload) {
+      return rawPayload;
+    }
+
+    serialize(snapshot, options) {
+      const serializedResource = {
+        id: snapshot.id(),
+        type: snapshot.modelName,
+        attributes: snapshot.attributes()
+      };
+
+      return serializedResource;
+    }
+  }
+ ```
+
+
+  #### Serializer Resolution
+
+  The instances of serializers defined in `app/serializers/` can be looked up
+  via `store.serializerFor(name)`.
+
+  It is recommended that applications define only a single `application` adapter and serializer
+  where possible.
+
+  `serializerFor` first attempts to find a serializer with an exact match on `name`,
+  then falls back to checking for the presence of a serializer named `application`.
+
+  ```ts
+  store.serializerFor('author');
+
+  // lookup paths (in order) =>
+  //   app/serializers/author.js
+  //   app/serializers/application.js
+  ```
+
+  Because most requests in `ember-data` occur from the perspective of a primary `type`
+  (or `modelName`) typically `serializerFor` will be used to find a serializer with a name
+  matching that of the primary resource `type` for the request, falling back to the `application`
+  serializer for those types that do not have a defined serializer. This is often described
+  as a `per-model` or `per-type` strategy for defining serializers. However, because APIs
+  rarely format payloads per-type but rather per-API-version, this may not be a desired strategy.
+
+  If you have multiple API formats and the per-type strategy is not viable, one strategy is to
+  write an `application` adapter and serializer that make use of `options` to specify the desired
+  format when making a request.
+
+  ### Using a Serializer
+
+  Any serializer in `app/serializers` can be looked up by `name` using `store.serializerFor(name)`.
+
+  ### Default Serializers
+
+  It is recommended that apps write their own serializer to best suit the needs of their API and
+  application.
+
+  However, for applications whose APIs are *very close to* or *exactly* the `REST` or `JSON:API`
+  format the `@ember-data/serializer` package contains implementations these applications can
+  extend. It also contains a simple `JSONSerializer` for serializing to/from very basic JSON objects.
```

Comment:
Putting myself in the shoes of someone who might take "recommendations" a bit too rigidly, I might have trouble reconciling the existence of these turnkey solutions with the above "we recommend you write your own serializer".